Java/core/src/test/java/com/amazon/randomcutforest/RandomCutForestTest.java

```diff
@@ -985,4 +986,64 @@ public void testOutOfOrderUpdate() {
         // the {20,-20} point is present still
         assertEquals(forest.getNearNeighborsInSample(new double[] { 20.0, -20.0 }, 1).size(), 1);
     }
+
+    @Test
+    public void testFloatingPointRandomCut() {
+        int dimensions = 16;
+        int numberOfTrees = 41;
+        int sampleSize = 64;
+        int dataSize = 4000 * sampleSize;
+        double[][] big = generateShingledData(dataSize, dimensions, 2);
+        RandomCutForest forest = RandomCutForest.builder().compact(true).dimensions(dimensions)
```

Comment:
the test would run at least half an hour, is that fine to do this as a unit test?

Reply:
It's about 6 or 7 seconds on a MAC ... try it.